include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -779,6 +779,28 @@ class BuildCallInstance : public calyx::FuncOpPartialLoweringPattern {
   ComponentOp getCallComponent(mlir::func::CallOp callOp) const;
 };
 
+/// Predicate information for the floating point comparisons
+struct PredicateInfo {
+  struct InputPorts {
+    // Relevant ports to extract from the `std_compareFN`. For example, we
+    // extract the `lt` and the `unordered` ports when the predicate is `oge`.
+    enum class Port { eq, gt, lt, unordered };
+    Port port;
+    // Whether we should invert the port before passing as inputs to the `op`
+    // field. For example, we should invert both the `lt` and the `unordered`
+    // port just extracted for predicate `oge`.
+    bool invert;
+  };
+
+  // The combinational logic to apply to the input ports. For example, we should
+  // apply `logicAnd` to the two input ports for predicate `oge`.
+  enum class CombLogic { logicAnd, logicOr, noComb };
```

Comment:
Suggestion: It might even be simpler to say None, i.e., `CombLogic::None` seems like a less verbose version of `CombLogic::noComb`. Usually the NONE case is placed first as well, since it corresponds to the integer 0.

Reply:
ah indeed, i was concerned `none` was a keyword or something